src/ensembl/src/content/app/custom-download/containers/content/filter-accordion/FiltersAccordion.tsx

```diff
@@ -7,32 +7,87 @@ import {
   AccordionItem,
   AccordionItemHeading,
   AccordionItemPanel,
-  AccordionItemButton
+  AccordionItemButton,
+  AccordionItemPermanentBlock
 } from 'src/shared/components/accordion';
 
-import styles from './FiltersAccordion.scss';
-
-import { getFiltersAccordionExpandedPanel } from '../../../state/filters/filtersSelector';
+import { getPreviewResult } from '../../../state/customDownloadSelectors';
+import {
+  getFiltersAccordionExpandedPanels,
+  getSelectedFilters
+} from '../../../state/filters/filtersSelector';
 import {
   setFiltersAccordionExpandedPanel,
   resetSelectedFilters,
   updateSelectedFilters
 } from '../../../state/filters/filtersActions';
 
-import customDownloadStorageService from 'src/content/app/custom-download/services/custom-download-storage-service';
 import JSONValue from 'src/shared/types/JSON';
 
-import { Genes } from './sections';
 import ImageButton, {
   ImageButtonStatus
 } from 'src/shared/components/image-button/ImageButton';
+import BadgedButton from 'src/shared/components/badged-button/BadgedButton';
+import { getCommaSeparatedNumber } from 'src/shared/helpers/numberFormatter';
 import { ReactComponent as ResetIcon } from 'static/img/shared/trash.svg';
+import FiltersAccordionSection from 'src/content/app/custom-download/containers/content/filter-accordion/sections/FiltersAccordionSection';
+import styles from './FiltersAccordion.scss';
+
+type Filter = {
+  [key: string]: boolean;
+};
+type SelectedFilters = {
+  [key: string]: boolean | string | string[] | Filter;
+};
+
+type StateProps = {
+  expandedPanels: string[];
+  selectedFilters: {};
+  preview: JSONValue;
+};
+
+type DispatchProps = {
+  setFiltersAccordionExpandedPanel: (expandedPanels: string[]) => void;
+  resetSelectedFilters: () => void;
+  updateSelectedFilters: (filters: JSONValue) => void;
+};
+
+const getTotalSelectedFilters = (
+  filters: SelectedFilters,
+  totalSelectedFilters = 0
+) => {
+  if (!filters) {
+    return 0;
+  }
+  Object.keys(filters).forEach((key: string) => {
+    if (key === 'preExpanded') {
+      // Skip preExpanded keys
+    } else if (typeof filters[key] === 'boolean' && filters[key] === true) {
+      totalSelectedFilters++;
+    } else if (typeof filters[key] === 'string' && filters[key] !== '') {
+      totalSelectedFilters++;
+    } else if (
+      Array.isArray(filters[key]) &&
+      (filters[key] as string[]).length > 0
+    ) {
+      totalSelectedFilters++;
+    } else if (typeof filters[key] === 'object') {
+      totalSelectedFilters = getTotalSelectedFilters(
```

Comment:
is this part correct? `= getTotalSelectedFilters`, not `+= getTotalSelectedFilters`?

Reply:
Yes, it is correct.